subprojects/snapshots/src/main/java/org/gradle/internal/snapshot/MissingFileSnapshot.java

```diff
@@ -49,4 +53,11 @@ public boolean isContentAndMetadataUpToDate(FileSystemLocationSnapshot other) {
     public void accept(FileSystemSnapshotVisitor visitor) {
         visitor.visitFile(this);
     }
+
+    public boolean isSymbolicLink() {
+        if (isSymbolicLink == null) {
```

Comment:
don't think we need synchronization here, right?

Reply:
We don't.